prow/scripts/provision-vm-and-start-cluster-users-k3d.sh

```diff
@@ -0,0 +1,118 @@
+#!/bin/bash
+
+# This script is designed to provision a new vm and start serverless chart. It takes an optional positional parameter using --image flag
+# Use this flag to specify the custom image for provisining vms. If no flag is provided, the latest custom image is used.
+
+set -o errexit
+
+date
+
+readonly SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+readonly TEST_INFRA_SOURCES_DIR="$(cd "${SCRIPT_DIR}/../../" && pwd)"
+
+# shellcheck source=prow/scripts/lib/log.sh
+source "${SCRIPT_DIR}/lib/log.sh"
+# shellcheck source=prow/scripts/lib/utils.sh
+source "$SCRIPT_DIR/lib/utils.sh"
+# shellcheck source=prow/scripts/lib/gcp.sh
+source "$SCRIPT_DIR/lib/gcp.sh"
+
+if [[ "${BUILD_TYPE}" == "pr" ]]; then
+    log::info "Execute Job Guard"
+    "${TEST_INFRA_SOURCES_DIR}/development/jobguard/scripts/run.sh"
+fi
+
+cleanup() {
+    ARG=$?
+    log::info "Removing instance kyma-integration-test-${RANDOM_ID}"
+    gcloud compute instances delete --zone="${ZONE}" "kyma-integration-test-${RANDOM_ID}" || true ### Workaround: not failing the job regardless of the vm deletion result
+    date
+    exit $ARG
+}
+
+function testCustomImage() {
+    CUSTOM_IMAGE="$1"
+    IMAGE_EXISTS=$(gcloud compute images list --filter "name:${CUSTOM_IMAGE}" | tail -n +2 | awk '{print $1}')
+    if [[ -z "$IMAGE_EXISTS" ]]; then
+        log::error "${CUSTOM_IMAGE} is invalid, it is not available in GCP images list, the script will terminate ..." && exit 1
+    fi
+}
+
+gcp::authenticate \
+    -c "${GOOGLE_APPLICATION_CREDENTIALS}"
+
+RANDOM_ID=$(openssl rand -hex 4)
+
+LABELS=""
+if [[ -z "${PULL_NUMBER}" ]]; then
+    LABELS=(--labels "branch=$PULL_BASE_REF,job-name=kyma-integration")
```

Comment:
This line will be problematic because `PULL_BASE_REF` assumes the name of a current branch. Some ref names have not allowed characters, specifically release branches which have dots in their name, which breaks the gcloud VM provision command. Please use the bash variables ability to replace the characters. 
```suggestion
    LABELS=(--labels "branch=${PULL_BASE_REF/./-},job-name=kyma-integration")
```